tests/test_OdmrLogic.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+
+"""
+This file contains unit tests for all qudi fit routines for exponential decay models.
+
+Copyright (c) 2021, the qudi developers. See the AUTHORS.md file at the top-level directory of this
+distribution and on <https://github.com/Ulm-IQO/qudi-core/>
+
+This file is part of qudi.
+
+Qudi is free software: you can redistribute it and/or modify it under the terms of
+the GNU Lesser General Public License as published by the Free Software Foundation,
+either version 3 of the License, or (at your option) any later version.
+
+Qudi is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with qudi.
+If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import os
+import time
+import math
+import numpy as np
+import coverage
+import pytest
+from qudi.util.network import netobtain
+
+MODULE = 'odmr_logic'
+BASE = 'logic'
+CHANNELS = ('APD counts', 'Photodiode')
+FIT_MODEL = 'Gaussian Dip'
+TOLERANCE = 10 # tolerance for signal data range
+
+
+def get_scanner(module):
+    return module._data_scanner()
+
+def get_microwave(module):
+    return module._microwave()
+
+def get_odmr_range(length, scanner):
+    """
+    Simulate odmr scan data and return signal data
+    Parameters
```

Comment:
Should have a newline before 'Parameters'. It's not yet part of the style guide for the repo but it will soon be